tools/logger/logger.c

```diff
@@ -362,6 +351,17 @@ int main(int argc, char *argv[])
 	if (isatty(fileno(config.out_fd)) != 1)
 		config.use_colors = 0;
 
+	if (config.version_fw) {
+		config.version_fd = fopen(config.version_file, "rb");
+		if (!config.version_fd) {
+			ret = errno;
+			fprintf(stderr,
+				"error: Unable to open version file %s: %s\n",
+				config.version_file, strerror(ret));
+			goto out;
+		}
+	}
```

Comment:
this is fine, but the commit message is somehow hard to digest:
"So when the driver is not loaded, we get the same (missing trace) error
trace message whether we use the -n option or not."

When the driver is not loaded then the entire `/sys/kernel/debug/sof` directory is non existent, so it does not matter which file you try to read, none of them will be present.

However, _if_ the dtrace is disabled in kernel then the `trace` and `filter` files are going to be missing and they will only appear if you reload the modules with `options snd_sof sof_debug=1` at minimum.

Reply:
> When the driver is not loaded then the entire /sys/kernel/debug/sof directory is non existent, so it does not matter which file you try to read, none of them will be present.

It matters a lot for the next commit but I tried to avoid a reference to the future.

> However, if the dtrace is disabled in kernel then the trace and filter files are going to be missing and they will only appear if you reload the modules with options snd_sof sof_debug=1 at minimum.

and? Not sure I see your point here sorry, is this a better commit message?

Reply:
@marc-hb, what about printing something like "driver is not loaded" when the driver is not loaded (the sof debugfs directory is not present along with the fw_version file)?

If the version file is optional then why we fail when it is not present?

Reply:
> If the version file is optional then why we fail when it is not present?

We don't, this code is not run when config.version_fw is false.

I moved the version check _after_ waiting for the trace file. I left the error handling because it does not hurt, opening could fail for some other reason (like a race)

There is a "waiting for /sys/..." message above.